keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -472,6 +484,21 @@ public DoubleTensor logInPlace() {
         return this;
     }
 
+    /**
+     * This is identical to log().times(y), except that it changes NaN results to 0.
+     * This is important when calculating 0log0, which should return 0
+     * See https://arcsecond.wordpress.com/2009/03/19/0log0-0-for-real/ for some mathematical justification
+     * @param y The tensor value to multiply by
+     * @return the log of this tensor multiplied by y
+     */
+    @Override
+    public DoubleTensor safeLogTimesInPlace(DoubleTensor y) {
+        TensorValidator.NAN_CATCHER.validate(this);
+        TensorValidator.NAN_CATCHER.validate(y);
+        DoubleTensor result = this.logInPlace().timesInPlace(y);
+        return TensorValidator.NAN_FIXER.validate(result);
```

Comment:
If `this` is not zero but `y` is, then we should still return `NaN`

Reply:
Oh right, I mistakenly thought `log(0)` returns `NaN`!

Reply:
👍